### DISCUSSION TITLE
Overlapping Relationship Lines

### DISCUSSION THREAD

**Discussion by rahuggins** [structurizr/ui#49] [General]
Another user asked the question below last year, but I have not been able to find where it was answered. I am running into the same issue and wanted to resurface this topic. 

I am trying to create an architecture diagram for a system(fairly large) that has multiple elements and dependencies, and using an automatic tb layout to render the diagram. In the rendered diagram, when we zoom in to container levels, the relations are displayed using overlapping lines which makes the diagrams a little cluttered and ambiguous.

I then tried to edit the diagram by disabling the automatic layout, and changing the origin point of relationship lines but that also did not work out since the relationships apparently start from the center of the element's boundary always. This can become problematic in cases there are let's say minimum 6-7 relationships originating from an element , the rendered lines will overlap then.

    Is there any option to disable overlapping lines?
    Is there any option to originate the relationships slightly away from the center of the element boundary so that it can be used to provide more clarity to the viewers in case of multiple relationships?
    On a side note, is there an option that relationship lines do not cross each other but apparently 'jump' over it (something like

**Comment by simonbrowndotje** [comment]
The origin points of relationships are fixed, but you can add vertices to the lines and move the labels; see https://docs.structurizr.com/ui/diagrams/editor#relationships for more.

**Reply by TheAdail** [reply]
I have two systems (A, B) with two subsystems each (A.1, A.2, B.1, B.2). A.1 -> B.2 and A.2 -> B.2.
In a Container diagram I can see only one of the relationships.
I tried adding vertices and moving labels, but either the lines/labels are moving together, or the line/label behind is not shown.
I am using the latest onpremises image from Docker Hub.

**Reply by pkunze** [reply]
I am facing the same problem in the dsl and structurizr lite. Is there a way to work around this while maintaining enables autoLayout?
